### PR TITLE
Bug fixes and performance improvements for DistributedStoreSingleFeatureSpace and split_idx function

### PR DESCRIPTION
**Pull request by felix0097**
**Description of changes**

* Added bug fix where `unknown` cell_type label would break subsetting based on cell type
* Added bug fix where having an array in the uns data would break subsetting, e.g. when one data set has more than one tissue type
* Sped up `get_subset_idx` function from `DistributedStoreSingleFeatureSpace` by replacing python code with corresponding numpy functions (please check whether this is really equivalent)
* Sped up `split_idx` function from `sfaira.estimators.keras.py` by replacing list comprehension with equivalent numpy functions
* Small changes to code layout

**Technical details**
Performance improvements are mostly from replacing list comprehensions with equivalent numpy functions

### CODE REVIEW

**Review by davidsebfischer — APPROVED** [review]

File: sfaira/data/store/single_store.py [hunk 1]
```diff
@@ -298,10 +300,10 @@ def get_idx(adata, obs, k, v, xv, dataset):
             # Cannot index on view here as indexing on view of views of backed anndata objects is not yet supported.
             idx_subset = get_idx(adata=adata_k, obs=obs_k, k=attr_key, v=values, xv=excluded_values, dataset=key)
             # Keep intersection of old and new hits.
-            idx_new = np.sort(list(set(np.asarray(idx_old).tolist()).intersection(
-                set(np.asarray(idx_subset).tolist()))))
+            idx_new = np.intersect1d(idx_old, idx_subset)
```
Comment (felix0097) [inline sfaira/data/store/single_store.py:303]: please check whether this is equivalent to above two lines

**Comment by felix0097** [comment]
All unit test pass - except for the two cellxgene ones again
